en/flying/terrain_following_holding.md

```diff
@@ -0,0 +1,81 @@
+
+> **Note** More precisely, the vehicle will use the *primary source of altitude data* as defined in [EKF2_HGT_MODE](../advanced_config/parameter_reference.md#EKF2_HGT_MODE).
+    This is, by default, the barometer.
+
+Terrain following is enabled by setting [MPC_ALT_MODE](../advanced_config/parameter_reference.md#MPC_ALT_MODE) to `1`.
+
+
+## Terrain Hold {#terrain_hold}
+
+Terrain holding uses a distance sensor to help a vehicle to better maintain a constant height above ground in [position](../flight_modes/position_mc.md) or [altitude](../flight_modes/altitude_mc.md) modes, when horizontally stationary at low altitude.
+This allows a vehicle to avoid altitude changes due to barometer drift or excessive barometer interference from rotor wash.
+
+> **Note** Terrain hold is enabled only in *multicopters* and *VTOL vehicles in MC mode*, and can be used in [position](../flight_modes/position_mc.md) and [altitude modes](../flight_modes/altitude_mc.md).
+
+When moving horizontally (`speed >` [MPC_HOLD_MAX_XY](../advanced_config/parameter_reference.md#MPC_HOLD_MAX_XY), or above the altitude where the distance sensor is providing valid data, the vehicle will instead use the *primary source of altitude data* ([EKF2_HGT_MODE](../advanced_config/parameter_reference.md#EKF2_HGT_MODE)).
+Typically this is the barometer.
+
+Terrain holding is enabled by setting [MPC_ALT_MODE](../advanced_config/parameter_reference.md#MPC_ALT_MODE) to `2`.
+
+> **Note** *Terrain hold* is implemented similarly to *terrain following.
+  It uses the output of the EKF estimator to provide the altitude estimate, and the estimated terrain altitude (calculated from distance sensor measurements using another estimator) to provide the altitude setpoint.
+  As the distance to ground changes due to external forces, the altitude setpoint adjusts to keep the height above ground constant.
+
+
+## Distance Sensor as Primary Source of Height {#distance_sensor_primary_altitude_source}
+
+PX4 allows you to make a distance sensor the *primary source of altitude data*.
+This may be useful when no barometer is available, or for applications when the vehicle is *guaranteed* to only fly over a near-flat surface (e.g. indoors).
+
+> **Tip** The default and preferred altitude sensor for most use cases is the barometer (when available).
+
+When using a distance sensor as the primary source of height, fliers should be aware:
+- Flying over obstacles can lead to the estimator rejecting data (due to internal data consistency checks).
+  This will result in TBD.
+  > **Note** A scenario where this can occur is when the vehicle is ascending a slope at a near-constant height above ground;
+    The measured/estimated altitude does not change, but consistency checks show an error against the barometer measurements.
+- The local NED origin will move up and down with ground level.
+- Rangefinder performance over uneven surfaces (e.g. trees) can be very poor, resulting in noisy and inconsistent data.
```

Comment:
Again, if we see noisy data presumably this means that we might have a rapidly varying estimate of position? What happens then ?

Reply:
@hamishwillee Poor altitude hold